### ISSUE TITLE
Custom JsonConverter never triggered?

### ISSUE THREAD

**Issue by spplante** [Azure/azure-functions-durable-extension#963]
### Description
I have an activity which returns an object that has one of it's properties defined with an interface type. To avoid getting the "***Type is an interface or abstract class and cannot be instantiated***" error upon deserialization, I created a simple JsonConverter and decorated the property with [JsonConverter(typeof(...))], yet my JsonConverter is never called when the framework tries to serialize/deserialize the output of my activity?

### Expected behavior
A JsonConverter specified using an attribute  decorator should be called as expected

### Actual behavior
A JsonConverter specified using an attribute decorator does not get called

### Relevant source code snippets
Here is the basic code that I can't get to work :

**Object returned from the activity**
```csharp
public class MySerializableObject : IMySerializableObject
{
    [JsonConverter(typeof(AbstractConverter<IMyObject, MyObject>))]
    public IMyObject Obj { get; set; }

    public MySerializableObject() { }
}
```

**Interface**
```csharp
public interface IMySerializableObject
{
        IMyObject Obj { get; }
}
```

**Simple JsonConverter which converts a property from TAbstract to TConcrete**
```csharp
public class AbstractConverter<TAbstract, TConcrete> : JsonConverter where TConcrete : TAbstract
{
    public override bool CanConvert(Type objectType)
    {
        return objectType == typeof(TAbstract);
    }

    public override Object ReadJson(JsonReader reader, Type type, Object value, JsonSerializer serializer)
    {
        return serializer.Deserialize<TConcrete>(reader);
    }

    public override void WriteJson(JsonWriter writer, Object value, JsonSerializer serializer)
    {
        serializer.Serialize(writer, value);
    }
}
```

**Comment by cgillum** [comment]
Does it work correctly if you use it in a standalone console app? If so, would you mind sharing a code snippet that demonstrates it working?

**Comment by spplante** [comment]
Yes I just did a console app and everything works as expected! Here is the functional
 console app.

[ConsoleApp1.zip](https://github.com/Azure/azure-functions-durable-extension/files/3699021/ConsoleApp1.zip)

Thanks!

**Comment by revoxover** [comment]
I recently ran into the same issue.

**Comment by dcousino** [comment]
Anything on this? This prevents us from being able to use interfaces as parameters.

**Comment by ConnorMcMahon** [comment]
Message serialization should now be controlled via injecting an `IMessageSerializerSettingsFactory`. We have an issue to track documenting it more clearly: #1208.

**Comment by seanix99** [comment]
Cool that you have added this interface, but how can we access it and register our custom converter if we are in function 1.0 and .Net 4.6? We can't use .net Core, so no dependencies injection.

**Comment by spplante** [comment]
x2 @ConnorMcMahon while v2.2.0 fixed the function v1 not being able to run orchestrator clients in v2.1.0 (fix #1258), **IMessageSerializerSettingsFactory** still does not work using function v1 since it requires .net Core...Any plan to make it work for function v1 with .net 4.6?

**Comment by ConnorMcMahon** [comment]
Unfortunately, Functions V1 has no formal dependency injection feature for us to interface with. Historically, we allowed customers to specify the name of a type that existed in their assembly to mimic this behavior, but it was rather fragile.

@cgillum, what would your thoughts be about us adding a subsection of the host.json that is dedicated for V1 dependency injection that works how we used to allow the NotificationHandler back in Durable 1.x?
